src/main/java/de/tum/cit/aet/artemis/iris/service/pyris/dto/data/PyrisLLMCostDTO.java

```diff
@@ -0,0 +1,6 @@
+package de.tum.cit.aet.artemis.iris.service.pyris.dto.data;
+
+import de.tum.cit.aet.artemis.core.domain.LLMServiceType;
+
+public record PyrisLLMCostDTO(String modelInfo, int numInputTokens, float costPerInputToken, int numOutputTokens, float costPerOutputToken, LLMServiceType pipeline) {
```

Comment:
For the same reason as in my other review, we might want to remove ``LLMServiceType`` here and use a String instead, just in case something that is unknown in our version of the enum gets passed. We should not crash the entire pipeline in this case. You can still translate the string to the enum value later when processing and saving the cost

Reply:
Good point, type of `ServiceType` is now String, and the method checks if `job` is not null before trying to access `jobId`